forum/views.py

```diff
@@ -1,48 +1,69 @@
-
-# User not logged in
-def index(request):
-    print "index view"
-    return HttpResponse("Index View Sir")
-
-# Page will exist for functionality and then I would add more complicated jquery 
-def reg_log(request):
-    print "log_reg view"
-    return render("forum/log_reg.html")
-
-# Would take in intial login and registration information as well as search ultimately/ much future state
-def post(request):
-    print "post view"
-    return redirect("/main")
-
-# Specific to the user once logged in
-def main(request, user_name):
-    print "main view"
-    return HttpResponse("main View Sir")
-
-# Would be standard form or html page with some user info already populated through a api pull
-def profile(request, user_name):
-    print "profile view"
-    return HttpResponse("profile View Sir")
-
-
-# Advanced feature
-def mail(request):
-    print "mail view"
-    return HttpResponse("mail View Sir")
-
-# Advanced feature.  Would be standard form or html page with some user info already populated through a api pull
-def preferences(request):
-    print "preferences view"
-    return HttpResponse("preferences View Sir")
-
-def logout(request):
-    print 'logging out'
-    request.session.clear()
-    return redirect('/')
+from django.shortcuts import render, redirect, HttpResponse, get_object_or_404
+from .forms import RegisterForm, LoginForm
+
+
+# Can change "standard query reference from object_list to whatever as below"
+class HomePageView(ListView):
+    template_name = "forum/list.html"
+    context_object_name= "post_list"
+
+    # Throwing error on my end for comment, but view works
+    #Should be first page
+    def get_queryset(self):
+        # Not sure how to pass below...need to refractor into class createview maybe?  Sending context forward as is creates error
+        vote_count = Post.objects.first().votes.count()
+        upvotes = Post.objects.first().votes.filter(vote=1).count()
+        downvotes = Post.objects.first().votes.filter(vote=-1).count()
+        total = upvotes - downvotes
+        context = {
+            "vc": vote_count,
+            "total":total
+        }
+        print(context)
+        return Post.objects.all()
+
+
+def posttest(request):
+    if request.method == "POST":
+        try:
+            user = request.user
+            post = Post.objects.get(id=request.POST["post_id"])
+            vote = request.POST["vote_choice"]
+            PostVote.objects.create(
+                user= user,
+                post= post,
+                vote= vote
+                )
+        except:
+            print("could not create")
+        return redirect("/")
+
+# Detail view containing some specific user data...could evolve to user preferences page
+class UserView(DetailView):
+    model = User
+
+    def get(self, request):
+        user = request.user
+        context = {
+            "user":user, 
```

Comment:
FYI, you don't *need* to pass in `user` to the view. You already have access to that in all templates by calling `{{ request.user }}`.

Reply:
Tried this in user but it does not seem to work for some reason.  Removed context pass.  Security element since user.is_logged_in may not be true?  Will need to test with you.
{{ request.user }} in HTML = Anonymous User
{{ user}} in HTML = Anonymous User
{{ user.first_name }} in HTML = blank
 {{ user.id }} in HTML = None

Reply:
That's expected behavior. If you're looking at a page and you're not logged in, Django defaults you to `AnonymousUser`. If you try `request.user.is_authenticated`, it'll tell you it's `False`.

If you want to test the templates with an authenticated user, you should look at it while you're a superuser.